dashboard/src/modules/components/OverviewComponent/NewRunsComponent.jsx

```diff
@@ -122,7 +119,17 @@ const NewRunsComponent = () => {
         ? [...otherExpandedRunNames, run.name]
         : otherExpandedRunNames;
     });
-  const isRunExpanded = (run) => expandedRunNames.includes(run.name);
+  const isRunExpanded = useCallback(
+    (run) => expandedRunNames.includes(run.name),
+    [expandedRunNames]
+  );
+  const updateTblValue = (newValue, metadata, rId) => {
+    dispatch(editMetadata(newValue, metadata, rId, "newRuns"));
+  };
```

Comment:
> Let's take this in new PR

Sounds good.

> It will be great if I get brief outline to address this problem.

I'll try to write something up in the next reply.

Reply:
`editMetadata()` needs to know whether we're operating on `"newRuns"` or `"savedRuns"`, not because it cares itself but because it calls `filterDatasetType()` and `updateDatasetType()` which care...so the question is, how can we change the code so that `editMetadata()` doesn't need to know?

One option is to pry the calls to `filterDatasetType()` and `updateDatasetType()` out of `editMetadata()` and hoist them into the caller.  In this particular case, that would be easy to do, since they are the first and last operations, respectively, in `editMetadata()`.

The other option is to reevaluate the requirements of `filterDatasetType()` and `updateDatasetType()` (and `editMetadata()`) and provide them with what they actually need instead of giving them hints on how to get it.

In this particular situation, I am inclined to do some of both.

The callers of `editMetadata()` know which set of runs they want to edit.  So, instead of passing in `type` which has to be translated by `filterDatasetType()`, we could promote `data` to be a parameter and have the caller pass in the appropriate value (i.e., the value of `getState().overview.initNewRuns` when the caller is `NewRunsComponent` or the value of `getState().overview.savedRuns` when the caller is `SavedRunsComponent`).  Doing so gets rid of the need for the `filterDatasetType()` function and its conditional operation, because we already know the answer from context when the call is made.

Similarly, the caller of `editMetadata()` effectively already knows what `TYPES` constant should be used in the dispatch at the end of editing the metadata, so passing an abstract type which has to be translated via a conditional into the appropriate `TYPES` constant is probably a waste, as well.  So, we could either have the caller provide the appropriate value (i.e., `TYPES.INIT_NEW_RUNS` when the caller is `NewRunsComponent` or `TYPES.SAVED_RUNS` when the caller is `SavedRunsComponent`), but it would be better still to just have `editMetadata()` return the `data` payload value which the caller could pass to `dispatch()` with the appropriate `TYPES` value.

The result is that `editMetadata()` "does one thing" -- it accepts the appropriate run list, mutates it as requested, and hands it back -- and neither it nor anything it calls is not involved in selecting the list or invoking `dispatch()`.  It puts more weight on the callers (they have to select the list and invoke `dispatch()` when the edit is done), but the total work is actually lighter, because all the same steps are done but without the conditional code (no ternaries! 😉).  (And, the unit testing, once we have that, should be easier as well, because the individual functions are better focused.)

Now, all that said, simple code rearrangements like what I'm suggesting here are not always the right answer.  Modularity and encapsulation are also extremely important.  However, they can usually be maintained without requiring control parameters by shifting responsibilities to the caller and passing in/out data instead of controls.